lib/schema-compilers.js

```diff
@@ -49,6 +49,17 @@ function ValidatorCompiler (externalSchemas, options, cache) {
   }
 
   return function ({ schema, method, url, httpPart }) {
+    // Ajv does not support compiling two schemas with the same
+    // id inside the same instance. Therefore if we have already
+    // compiled the schema with the given id, we just return it.
+    if (schema.$id) {
+      const stored = ajv.getSchema(schema.$id)
+      if (stored) {
+        // TODO we might want to throw if two schemas have
+        // the same id but different content.
```

Comment:
When would this be true? How would we even test it? If someone is doing this, they are violating the spec.